dynamic_screenshots/generate_images.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+import os
+from playwright.async_api import async_playwright
+from tqdm import tqdm
+from utils import remove_footer
+from images import IMAGES
+
+async def login(page, username: str, password: str):
+    # Replace with your app's login URL and adjust selectors as needed.
+    await page.goto("http://127.0.0.1:4000/")
+    await page.get_by_text("Start now").click()
+    await page.get_by_label("Email address").fill(username)
+    await page.get_by_label("Password", exact=True).fill(password)
+    await page.get_by_role("button", name="Log in").click()
+    await page.wait_for_url("**/users/organisations/new", timeout=1000)
+    await page.pause()
+    await page.get_by_role("heading", name="SAIS Organisation 1 (R1L)").click()
+    await page.wait_for_url("**/dashboard", timeout=1000)
+
+
+async def main():
+    username = "nurse.joy@example.com"
+    password = "nurse.joy@example.com"
+    base_url = "http://127.0.0.1:4000/"
+
+    dynamic_screenshots_dir = os.path.dirname(__file__)
+    os.makedirs(os.path.join(dynamic_screenshots_dir,"screenshots"), exist_ok=True)
+
+    image_metadata = IMAGES
+
+    for image in tqdm(image_metadata):
+        tqdm.write(f"Capturing screenshot: {image['image_name']}")
+        target_url = base_url + image["url_extension"]
+        output_file = f"{dynamic_screenshots_dir}/../app/assets/images/{image['image_name']}"
+
+        full_page = image["full_page"] if "full_page" in image.keys() else False
+
+        async with async_playwright() as p:
+            browser = await p.chromium.launch()  # Set headless=True as needed
+            context = await browser.new_context(viewport={
+                "width": image["screen_size"]["width"],
+                "height": 100 if full_page else image["screen_size"]["height"]
+            })
+            page = await context.new_page()
+
+            # Login to the application
+            if "login" in image.keys():
```

Comment:
If this code path isn't being used right now, then you can safely remove it in another commit -- it'll be in version control. Best not to keep code paths that aren't being used as it can be confusing.